docs/manual/scala/guide/cluster/code/docs/home/scaladsl/persistence/CassandraBlogEventProcessor.scala

```diff
@@ -0,0 +1,95 @@
+package docs.home.scaladsl.persistence
+
+//#imports
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import akka.Done
+import com.datastax.driver.core.BoundStatement
+import com.datastax.driver.core.PreparedStatement
+import com.lightbend.lagom.scaladsl.persistence.AggregateEventTag
+import com.lightbend.lagom.scaladsl.persistence.EventStreamElement
+import com.lightbend.lagom.scaladsl.persistence.ReadSideProcessor
+import com.lightbend.lagom.scaladsl.persistence.cassandra.CassandraReadSide
+import com.lightbend.lagom.scaladsl.persistence.cassandra.CassandraSession
+
+//#imports
+
+trait CassandraBlogEventProcessor {
+
+  trait Initial {
+    //#initial
+    class BlogEventProcessor(session: CassandraSession, readSide: CassandraReadSide)(implicit ec: ExecutionContext)
+      extends ReadSideProcessor[BlogEvent] {
+
+      override def buildHandler(): ReadSideProcessor.ReadSideHandler[BlogEvent] = {
+        // TODO build read side handler
+        ???
+      }
+
+      override def aggregateTags: Set[AggregateEventTag[BlogEvent]] = {
+        // TODO return the tag for the events
+        ???
+      }
+    }
+    //#initial
+  }
+
+  class BlogEventProcessor(session: CassandraSession, readSide: CassandraReadSide)(implicit ec: ExecutionContext)
+    extends ReadSideProcessor[BlogEvent] {
+
+    //#tag
+    override def aggregateTags: Set[AggregateEventTag[BlogEvent]] =
+      BlogEvent.Tag.allTags
+    //#tag
+
+    //#create-table
+    private def createTable(): Future[Done] =
+      session.executeCreateTable("CREATE TABLE IF NOT EXISTS blogsummary ( " +
+        "id TEXT, title TEXT, PRIMARY KEY (id))")
+    //#create-table
+
+    //#prepare-statements
+    private var writeTitle: PreparedStatement = null // initialized in prepare
+
+    private def prepareWriteTitle(): Future[Done] = {
+      session.prepare("INSERT INTO blogsummary (id, title) VALUES (?, ?)")
+        .map { ps =>
+          this.writeTitle = ps
```

Comment:
Something I thought about already earlier but I don't think I noted it anywhere, mustn't `writeTitle` be `@volatile` ?

Given that side-effecting in `Future.map` like this is something of an antipattern, could we have an api that covered this use case better?